Doppler.HtmlEditorApi/Controllers/TemplatesController.cs

```diff
@@ -7,144 +7,52 @@
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.Extensions.Logging;
 using Doppler.HtmlEditorApi.Model;
+using Doppler.HtmlEditorApi.Infrastructure;
 
 
 namespace Doppler.HtmlEditorApi.Controllers
 {
     [Authorize]
     [ApiController]
-    public class TemplatesController : Controller
+    public class TemplatesController
     {
+        private readonly IRepository _repository;
+
+        public TemplatesController(IRepository Repository)
+        {
+            _repository = Repository;
+        }
+
         [Authorize(Policies.OWN_RESOURCE_OR_SUPERUSER)]
         [HttpGet("/accounts/{accountName}/templates/{templateId}")]
-        public ActionResult<TemplateModel> GetTemplate(string accountName, int templateId)
+        public async Task<ActionResult<TemplateModel>> GetTemplate(string accountName, int templateId)
         {
-            TemplateModel template = new()
-            {
-                name = "Template Name",
-                counters = new()
-                {
-                    u_row = 1,
-                    u_column = 2,
-                    u_content_text = 1,
-                    u_content_heading = 1,
-                    u_content_menu = 1
-                },
-                body = new()
-                {
-                    rows = Array.Empty<string>(),
-                    values = new()
-                    {
-                        textColor = "#000000",
-                        backgroundColor = "#e7e7e7",
-                        backgroundImage = new()
-                        {
-                            url = "",
-                            fullWidth = true,
-                            repeat = false,
-                            center = true,
-                            cover = false
-                        },
-                        contentWidth = "500px",
-                        contentAlign = "center",
-                        fontFamily = new()
-                        {
-                            label = "Arial",
-                            value = "arial,helvetica,sans-serif"
-                        },
-                        preheaderText = "",
-                        linkStyle = new()
-                        {
-                            body = true,
-                            linkColor = "#0000ee",
-                            linkHoverColor = "#0000ee",
-                            linkUnderline = true,
-                            linkHoverUnderline = true
-                        },
-                        _meta = new()
-                        {
-                            htmlID = "u_body",
-                            htmlClassNames = "u_body"
-                        }
-                    }
-                },
-                schemaVersion = 6
-            };
-
-            return template;
+            var template = await _repository.GetTemplateModel(accountName, templateId);
+            return new OkObjectResult(template);
```

Comment:
Acá creo que es mejor quitar el `OkObjectResult` para asegurarnos que el compilador verifique nuestros tipos.

```suggestion
            return template;
```

Reply:
Corregido